api/kyverno/v1/common_types.go

```diff
@@ -195,6 +195,18 @@ type ResourceFilter struct {
 
 // Mutation defines how resource are modified.
 type Mutation struct {
+
+	// mutateExisting controls whether to mutate existing resource ONLY
+	// The existing resources will be mutated ONLY if set to "true".
+	// Otherwise all resources including admission requests are mutated.
+	// Optional. Defaults to "false" if not specified.
+	// +optional
+	MutateExisting bool `json:"mutateExisting,omitempty" yaml:"mutatingExisting,omitempty"`
+
+	// Targets defines the target resources to be mutated.
+	// +optional
+	Targets []*TargetMutation `json:"targets,omitempty" yaml:"targets,omitempty"`
```

Comment:
Why a pointer ?

Reply:
Good question! 

Digging more into it and seems like there are pros and cons of using "a slice of structs" VS "a slice of pointers to structs".

- choose "slices of structs" when the struct is relatively small and you don't need to change its value
- choose "slices of pointers" when you need to change the value of each item inside the slice

In our case, though we need to range over the `targets`, there are still advantages of using a slice of structs including allocation savings considering `TargetMutation` only consists of four-string attributes.

Some useful references I found:
- [Bad Go: slices of pointers
](https://medium.com/@philpearl/bad-go-slices-of-pointers-ed3c06b8bb41)
- [Potential dangers of storing structs rather than pointers](https://medium.com/@opto_ej/there-are-other-nuances-one-should-consider-c798f12be15c)
- [The Slices of Structs or Slices of Pointers to Structs Dilemma](https://www.reddit.com/r/golang/comments/2yfo8i/the_slices_of_structs_or_slices_of_pointers_to/)

Reply:
Updated!